src/cockpit/channels/filesystem.py

```diff
@@ -457,7 +475,15 @@ def get_group(gid: int) -> 'str | int':
 
         def get_attrs(fd: int, name: str, follow: Follow) -> 'JsonDict | None':
             try:
-                buf = os.stat(name, follow_symlinks=follow.value, dir_fd=fd) if name else os.fstat(fd)
+                file_fd = Handle.open(name, os.O_PATH | os.O_NOFOLLOW, os.O_RDONLY, dir_fd=fd) if name else fd
```

Comment:
Why this change?  If you wanted to share the fd with faccess that would be one thing but you don't...